src/gateways/S3-Gateway.test.ts

```diff
@@ -0,0 +1,80 @@
+import S3Gateway from './S3-Gateway';
+
+describe('S3Gateway', () => {
+  let client;
+  beforeEach(() => {
+    client = {
+      putObject: jest.fn(() => ({ promise: jest.fn() })),
+      createPresignedPost: jest.fn((options, callback) =>
+        callback(null, {
+          url: 'https://s3.eu-west-2.amazonaws.com/bucketName',
+          fields: {
+            'X-Amz-Algorithm': 'AWS4-HMAC-SHA256',
+          },
+        })
+      ),
+    };
+  });
+
+  it('can create metadata document in S3', async () => {
+    const metadata = {
+      firstName: 'Andrew',
+      dob: '1999-09-09',
+      documentId: '123',
+    };
+
+    const expectedObject = {
+      Bucket: 'testBucket',
+      Key: `${metadata.documentId}/${metadata.documentId}.json`,
```

Comment:
Me and @kosiakkatrina were chatting about keys in another PR, I wonder if we need the document id to be repeated or if we could just save it as `{documentId}/metadata.json` instead?